authz/audit/stdout/stdout_logger.go

```diff
@@ -0,0 +1,100 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package stdout defines an stdout audit logger.
+package stdout
+
+import (
+	"encoding/json"
+	"log"
+	"time"
+
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/grpc/grpclog"
+)
+
+var grpcLogger = grpclog.Component("authz-audit")
+
+func init() {
+	audit.RegisterLoggerBuilder(&loggerBuilder{})
+}
+
+type event struct {
+	FullMethodName string `json:"fullMethodName"`
+	Principal      string `json:"principal"`
+	PolicyName     string `json:"policyName"`
+	MatchedRule    string `json:"matchedRule"`
+	Authorized     bool   `json:"authorized"`
+	Timestamp      string `json:"timestamp"` // Time when the audit event is logged via Log method
+}
+
+// logger implements the audit.Logger interface by logging to standard output.
+type logger struct {
+}
+
+// Log marshals the audit.Event to json and prints it using log.go
+func (logger *logger) Log(event *audit.Event) {
+	jsonBytes, err := json.Marshal(convertEvent(event))
+	if err != nil {
+		grpcLogger.Errorf("failed to marshal AuditEvent data to JSON: %v", err)
+		return
+	}
+	log.Println(string(jsonBytes))
+}
+
+const (
+	stdName = "stdout"
```

Comment:
Prefer `stdout_logger` for consistency with my PR.

Reply:
Yes, registered names should match across implementations. Is the name defined in a gRFC or design doc?

Reply:
https://github.com/grpc/proposal/pull/346 has the example authz policy config (l298) that has the name. The current revision is not entirely up to date, mostly for the language API sections. But this `stdout_logger` has been there for a while and no one has raised any objection. If there is no other concerns that I have not been aware of, I'd prefer sticking to `stdout_logger`.

Reply:
I have no concerns about the name. Just wanted to make sure that the Go implementation is in line with the gRFC and other implementations. Thanks for pointing this out. I was reviewing mostly for style and readability, not for correctness (but that creeps in as well).

Reply:
@easwars yes, Luwei reviews it primarily from gRFC / business logic correctness perspective so it's perfectly fine for you to review style/readability/overall fit for gRPC Go